File: Myecoprint_admin/ncej_admin/lib/data/module/informations_contact.dart

class InformationContact {
  int infoContactId;
  int storeId;
  String? email;
  String landPhone;
  DateTime? createdAt;
  DateTime? updatedAt;
  DateTime? deletedAt;

  InformationContact({
    required this.infoContactId,
    required this.storeId,
    this.email,
    required this.landPhone,
  
  });

  factory InformationContact.fromJson(Map<String, dynamic> json) {
    return InformationContact(
      infoContactId: json['info_contact_id']as int? ??0,
      storeId: json['store_id']as int? ??0,
      email: json['email']as String? ?? '',
      landPhone: json['land_phone']as String? ?? '',

    );
  }

  Map<String, dynamic> toJson() {
    return {
      'info_contact_id': infoContactId,
      'store_id': storeId,
      'email': email,
      'land_phone': landPhone,
 
    };
  }
}

File: backend_Myecoprint/backend_myecoprint/models/informations_contact.js

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Store= require('./store');
const informationContact = sequelize.define('informationContact ', {
    info_contact_id: {
        type: DataTypes.INTEGER,
        allowNull: false,
        primaryKey: true,
        autoIncrement: true,
    },
    store_id: {
        type: DataTypes.INTEGER,
        allowNull: false,
    },

    email: {
        type: DataTypes.STRING,
    },
    land_phone: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    createdAt: {
        type: DataTypes.DATE,
        allowNull: true,
    },
    updatedAt: {
        type: DataTypes.DATE,
        allowNull: true,
    },
    deletedAt: {
        type: DataTypes.DATE,
        allowNull: true,
    },
}, {
    timestamps: false, 
    paranoid: true, 
    tableName: 'informations_contacts', 
  });
  informationContact.belongsTo(Store, { foreignKey: 'store_id', as: 'stores' });
module.exports = informationContact;































